controllers/employees.controller.js

```diff
@@ -0,0 +1,80 @@
+const Employee = require('../models/employees.model');
+
+exports.getAll = async (req, res) => {
+  try {
+    res.json(await Employee.find().populate('department'));
+  } catch (err) {
+    res.status(500).json({ message: err });
+  }
+};
+
+exports.getRandom = async (req, res) => {
+  try {
+    const count = await Employee.countDocuments();
+    const rand = Math.floor(Math.random() * count);
+    const dep = await Employee.findOne().skip(rand).populate('department');
+    if (!dep) res.status(404).json({ message: 'Not found' });
+    else res.json(dep);
+  } catch (err) {
+    res.status(500).json({ message: err });
+  }
+};
+
+exports.getItem = async (req, res) => {
+  try {
+    const dep = await Employee.findById(req.params.id).populate('department');
+    if (!dep) res.status(404).json({ message: 'Not found' });
+    else res.json(dep);
+  } catch (err) {
+    res.status(500).json({ message: err });
+  }
+};
+
+exports.postItem = async (req, res) => {
+  try {
+    const { firstName, lastName, department } = req.body;
+    const newEmployee = new Employee({
+      firstName: firstName,
+      lastName: lastName,
+      department: department,
+    });
+    await newEmployee.save();
+    res.json({ message: 'OK' });
+  } catch (err) {
+    res.status(500).json({ message: err });
+  }
+};
+
+exports.putItem = async (req, res) => {
+  try {
+    const { firstName, lastName, department } = req.body;
+    const dep = await Employee.findById(req.params.id).populate('department');
```

Comment:
Nie musisz tutaj robić tego populate, jako, że do niczego tego `dep` nie używasz poza sprawdzeniem czy istnieje. Szczerze, to nawet nie ma potrzeby, abyś to w ogóle sprawdzał. Podajesz warunki po jakich ma aktualizować (tutaj po _id) i jak znajdzie to zaktualizuje, jak nie to nie.